server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -284,12 +288,14 @@ public DateFieldMapper build(ContentPath contentPath) {
 
     public static final TypeParser MILLIS_PARSER = new TypeParser((n, c) -> {
         boolean ignoreMalformedByDefault = IGNORE_MALFORMED_SETTING.get(c.getSettings());
-        return new Builder(n, Resolution.MILLISECONDS, c.getDateFormatter(), ignoreMalformedByDefault, c.indexVersionCreated());
+        DateFormatter formatter = c.getDateFormatter() != null ? c.getDateFormatter() : DEFAULT_DATE_TIME_FORMATTER;
```

Comment:
Could you explain why this logic needs to change?

Reply:
That's a leftover from another approach. Good catch ! I removed it in https://github.com/elastic/elasticsearch/pull/70463/commits/5e5b85cf6bbed37c674708a198279572602cfad0